test/chat_api_web/controllers/note_controller_test.exs

```diff
@@ -0,0 +1,88 @@
+defmodule ChatApiWeb.NoteControllerTest do
+  use ChatApiWeb.ConnCase
+
+  alias ChatApi.Notes
+  alias ChatApi.Notes.Note
+
+  @create_attrs %{
+    body: "some body"
+  }
+  @update_attrs %{
+    body: "some updated body"
+  }
+  @invalid_attrs %{body: nil}
+
+  def fixture(:note) do
+    {:ok, note} = Notes.create_note(@create_attrs)
+    note
+  end
+
+  setup %{conn: conn} do
+    {:ok, conn: put_req_header(conn, "accept", "application/json")}
+  end
+
+  describe "index" do
+    test "lists all notes", %{conn: conn} do
+      conn = get(conn, Routes.note_path(conn, :index))
+      assert json_response(conn, 200)["data"] == []
+    end
+  end
+
+  describe "create note" do
+    test "renders note when data is valid", %{conn: conn} do
+      conn = post(conn, Routes.note_path(conn, :create), note: @create_attrs)
+      assert %{"id" => id} = json_response(conn, 201)["data"]
+
+      conn = get(conn, Routes.note_path(conn, :show, id))
+
+      assert %{
+               "id" => id,
+               "body" => "some body"
+             } = json_response(conn, 200)["data"]
+    end
+
+    test "renders errors when data is invalid", %{conn: conn} do
+      conn = post(conn, Routes.note_path(conn, :create), note: @invalid_attrs)
+      assert json_response(conn, 422)["errors"] != %{}
+    end
+  end
+
+  describe "update note" do
+    setup [:create_note]
+
+    test "renders note when data is valid", %{conn: conn, note: %Note{id: id} = note} do
+      conn = put(conn, Routes.note_path(conn, :update, note), note: @update_attrs)
+      assert %{"id" => ^id} = json_response(conn, 200)["data"]
+
+      conn = get(conn, Routes.note_path(conn, :show, id))
+
+      assert %{
+               "id" => id,
+               "body" => "some updated body"
+             } = json_response(conn, 200)["data"]
+    end
+
+    test "renders errors when data is invalid", %{conn: conn, note: note} do
+      conn = put(conn, Routes.note_path(conn, :update, note), note: @invalid_attrs)
+      assert json_response(conn, 422)["errors"] != %{}
+    end
+  end
+
+  describe "delete note" do
+    setup [:create_note]
+
+    test "deletes chosen note", %{conn: conn, note: note} do
+      conn = delete(conn, Routes.note_path(conn, :delete, note))
+      assert response(conn, 204)
+
+      assert_error_sent 404, fn ->
+        get(conn, Routes.note_path(conn, :show, note))
+      end
+    end
+  end
+
+  defp create_note(_) do
+    note = fixture(:note)
+    %{note: note}
+  end
+end
```

Comment:
let's get all the backend stuff fixed with tests passing before we worry about the UX :)